### PR TITLE
Add alerts for delayed studies

### LINKED ISSUE #4: Alerts
This ticket describes the features that will be provided for receiving alerts about studies.

### Research Managers
- [x] Will receive emails when ERB approval for any of their studies is about to expire
- [x] Will receive emails when any of their studies is delayed in completing
- [x] Will receive emails when any of their studies is delayed in receiving a response from the ERB

### PIs
- [x] Will receive emails when ERB approval for any of their studies is about to expire
- [x] Will receive emails when any of their studies is delayed in completing
- [x] Will receive emails when any of their studies is delayed in receiving a response from the ERB

### Administrators
- [x] Will receive emails when ERB approval for any study is about to expire
- [x] Will receive emails when any study is delayed in completing
- [x] Will receive emails when any study is delayed in receiving a response from the ERB

### LINKED ISSUE #69: Delays and Deadlines
There are several key dates in a study's lifecycle and then deadlines which arise from them. These need to be built into the system so that we can trigger alerts from them (See #4).

The current triggers are:
- [ ] Studies running past their expected completion time 
- [ ] ERB approval about to expire
- [ ] Studies which have been awaiting an ERB response for more than 3 months (e.g. `erb_status` is `submitted` and the `erb_submitted` date is older than 3 months)

### PR DESCRIPTION
**Pull request by stevenday**
This PR:
- Adds an "expected completion date" field to studies so that we can tell when
  a study should have completed
- Adds some new scopes to find studies which need to trigger alerts for being
  delayed, having erb approval expiring and being overdue an erb response.
- Adds a new model to track which alerts have been sent to who
- Adds mailers and a helper class to send out alerts
- Adds rake tasks to send alerts

We'll need to set up a cron config to run the rake tasks on the live site.

Closes #69
Closes #4

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -45,13 +45,17 @@ class Application < Rails::Application
       config.x.send("#{key}=".to_sym, value)
     end
 
-    # Set default_url_options for ActionMailer
+    # Set default_url_options and default from for ActionMailer
     if config.x.hostname.present? && config.x.port.present?
       config.action_mailer.default_url_options = { host: config.x.hostname,
                                                    port: config.x.port }
+      from = "do-not-reply@#{config.x.hostname}:#{config.x.port}"
+      ActionMailer::Base.default from: from
```
Comment (davea) [inline config/application.rb]: In the Devise config we're using `Rails.configuration.x.from_email` (from general.yml), can we do the same here?

**Comment by davea** [comment]
:+1:

File: config/application.rb (hunk 1 above)
Comment (stevenday) [inline config/application.rb]: ooh, so we are - I must have missed that - I'll change it

**Comment by stevenday** [comment]
Regarding putting things into the config - I figured that whilst they might change, they're only going to change globally, not for different versions of the app (e.g. staging, prod, etc) so having to change them in the model is as good as having to change them in some config file. The only really valuable thing would be to put them in the db instead, so that MSF can change them, but we don't really have time for that at the moment.

**Pull request merged**